source/module_lr/potentials/pot_hxc_lrtd.cpp

```diff
@@ -37,7 +37,7 @@ namespace LR
                 const std::set<std::string> lda_xc = { "lda", "pwlda" };
                 assert(lda_xc.count(this->xc_kernel_));
                 const int n_component = (1 == nspin) ? 1 : 3;
-                std::vector<double> v2rho2(n_component * nrxx);
+                this->xc_kernel_components_.v2rho2.resize(n_component * nrxx);
```

Comment:
I am confused about the name "components". Why there is a class instantiate with this name? what is the ownership and data manipulating priority design on "this" and "xc_kernel_components"?

Reply:
`KernelXC xc_kernel_componets_` is a private menber of `PotHxcLR` storeing the something like $\delta^2E/\delta\rho^2$,  $\delta^2E/\delta\rho\delta\sigma$  and $\delta^2E/\delta\sigma^2$. They will be frequently visited in calculate  in `PotHxcLR::cal_v_eff` to calculate integrals ($\delta E/\delta\rho$).

 `xc_kernel_componets_` is the only one instance of `KernelXC`. What's the best design from a performance perspective?
- to make them 2 independent classes as now? 
- to make `KernelXC` a nested class inside the private part of `PotXCLR`?
- to merge them into one class?

- Will the encapsulation, that is, to frequently call `xc_kernel_components_.v2rho2` instead of `v2rho2`, lead to a loss of performance? @caic99

Reply:
@maki49 according to your explanation, I will call the class `KernelXC` itself does not have complete abilities of taking care and manipuating data it owns, therefore it is not a sucessful design. The same comment also applies to some classes like `Orb_Control`.

Reply:
I've redesigned `KernelXC` in the newest commit, now the constructor sets everything and `PotHxcLR` only gets their const references.